arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1003,71 +1100,26 @@ void ClusterInfo::loadPlan() {
 
           continue;
         }
+          
+        bool const isBuilding = isCoordinator &&
+            arangodb::basics::VelocyPackHelper::getBooleanValue(collectionSlice,
+                                                                StaticStrings::AttrIsBuilding,
+                                                                false);
 
         auto const collectionId = collectionPairSlice.key.copyString();
-
+        
+        // check if we already know this collection (i.e. have it in our local cache).
+        // we do this to avoid rebuilding LogicalCollection objects from scratch in
+        // every iteration
+        // the cache check is very coarse-grained: it simply hashes the Plan VelocyPack
+        // data for the collection, and will only reuse a collection from the cache if 
+        // the hash is identical. 
+        auto [newCollection, hash] = buildCollection(isBuilding, existingCollections, collectionId, collectionSlice, *vocbase, newPlanVersion);
+        TRI_ASSERT(newCollection != nullptr);
```

Comment:
There is one actual change, in that an actually changed LogicalCollection object does no longer copy over index estimates from the old one. Is this intentional?

Reply:
That was actually intentional. Because _changes_ of existing collections should happen only rarely, and if they do, they may be because of index additions or removals. And in this case it is probably good to refetch estimates.